master/internal/db/postgres.go

```diff
@@ -1325,21 +1325,26 @@ func (db *PgDB) AddStep(step *model.Step) error {
 	if !step.IsNew() {
 		return errors.Errorf("unexpected state for new step: %v", step)
 	}
-	trial, err := db.TrialByID(step.TrialID)
+	var count int
+	err := db.namedGet(&count, `
```

Comment:
We should just add a uniqueness constraint to this column set in the database instead of trying to manage the integrity manually (in hindsight, we should've added this with your last migration). I'm not saying we _have_ to do that, but that if you see a bug with two steps at the same count, this would be it. But I think we should since if the trial has a bug where it sends the same step out back to back or something, we actually may see this.

Reply:
removed this since we already have the uniqueness constraint in the DB

Reply:
Forgot it already existed.